pkgs/applications/editors/vscode/extensions/default.nix

```diff
@@ -821,9 +824,24 @@ let
         mktplcRef = {
           name = "phptools-vscode";
           publisher = "devsense";
-          version = "1.33.12924";
-          sha256 = "sha256-ImaGkIe+MTO/utfVh3Giu0+jTSN0mmhgg6LvOod1suE=";
+          version = "1.33.12980";
+          sha256 = "sha256-wypPXiykubY33Ve/EMgmbvQe+KYaY70BZsZW77GuqDs=";
+          arch = "linux-x64";
```

Comment:
I think we should either set different `sha256` and `arch` based on `stdenv.system`, or we limit `meta.platforms = with lib.platforms [ x86_64-linux ]`. We should set `meta.platforms` either way, because the package contains native extensions which only work  on platforms they're build for

Reply:
The extension support many architectures, see 
![image](https://user-images.githubusercontent.com/252042/233796998-dbd9b61c-fd5a-4d1d-94e4-d0e67c4e4498.png)

What I don't understand is the following. If you remove the attribute `arch = "linux-x64";`,  then `buildVscodeMarketplaceExtension` will not download the adequate extension based on the current platform.

How are we supposed to fix that?!

Reply:
```nix
let
sources = {
 "x86_64-linux" = {
    arch = "linux-x86"; url = "..."; hash = "...";
  };
};
in sources.${stdenv.system}
```


But look for more detailed examples elsewhere in nixpkgs

Reply:
Ok, that I can do.

But what method would you suggest to compute the hashes of each extension for each architecture ?

Reply:
Bruteforce way is to download each one of them using nix prefetch url

Reply:
Done!